lib/Zonemaster/CLI.pm

```diff
@@ -839,16 +847,46 @@ sub add_fake_delegation {
 
         if ( $ip ) {
             my $net_ip = Net::IP::XS->new( $ip );
-            if ( ( $ip =~ /($IPV4_RE)/ && Net::IP::XS::ip_is_ipv4( $ip ) )
+            unless (
+                ( $ip =~ /($IPV4_RE)/ && Net::IP::XS::ip_is_ipv4( $ip ) )
                 or
-                 ( $ip =~ /($IPV6_RE)/ && Net::IP::XS::ip_is_ipv6( $ip ) )
-            ) {
-                push @{ $data{ $name } }, $ip;
-            }
-            else {
+                ( $ip =~ /($IPV6_RE)/ && Net::IP::XS::ip_is_ipv6( $ip ) )
+            )
+	    {
```

Comment:
I suggest to call Zonemaster::Engine::Validation::validate_ipv4 and Zonemaster::Engine::Validation::validate_ipv6, respectively, instead.

https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Validation.pm#L28-L57